core/applicationstate.h

```diff
@@ -0,0 +1,21 @@
+// SPDX-License-Identifier: GPL-2.0
+#ifndef APPLICATIONSTATE_H
+#define APPLICATIONSTATE_H
+
+// By using an enum class, the enum entries don't polute the global namespace.
+// Moreover, they are strongly typed. This means that they are not auto-converted
+// to integer types if e.g. used as array-indices.
+enum class ApplicationState {
+	Default,
+	EditDive,
+	PlanDive,
+	EditPlannedDive,
+	EditDiveSite,
+	FilterDive,
+	Count
+};
```

Comment:
I don't think we are consistent with this - don't we use all caps for things like this in other cases? For example in `enum cylinderuse`...

Looking further down I realize why you are using mixed case. This makes the changes smaller (or at least easier). I still think all caps would be better.

Reply:
The reason I didn't got wall all-caps is that normal enums work more like macros - the values are valid in the outer scope. I was under the impression that that's why typically all-caps are used: to avoid name conflicts. Here it's more like member-variables.

But I can change to all-caps, no problem. Up to you.

Reply:
That makes sense. I'm fine with it.